charts/spire/charts/spire-agent/templates/configmap.yaml

```diff
@@ -150,7 +150,9 @@ health_checks:
 telemetry:
   - Prometheus:
       - host: "0.0.0.0"
+        {{- if ne (int .port) 0 }}
         port: {{ .Values.telemetry.prometheus.port }}
+        {{- end }}
```

Comment:
```suggestion
        port: {{ .Values.telemetry.prometheus.port | int }}
```

Reply:
This suggestion would configure a default port of `0` if someone specifies a port that cannot be cast to int, e.g. the string "8000" instead of number `8000`
I personally prefer to not put the port if someone does not specify an int instead of defaulting it to `0`, what do you think?

Reply:
Could add a second line around it to fail the deploy if it isn't an int?

Reply:
There are other places in the PR that just cast to int as well. Should there be some consistency there?

Reply:
I wanted to be consistent with the existing port validation implemented in the same file [in L87](https://github.com/spiffe/helm-charts-hardened/blob/main/charts/spire/charts/spire-agent/templates/configmap.yaml#L87) and [in L90](https://github.com/spiffe/helm-charts-hardened/blob/main/charts/spire/charts/spire-agent/templates/configmap.yaml#L90).
In the daemonset.yaml I used `int` without this surrounding if statement. So for consistency, how should we deal with it? Go with the simple `| int` cast for ports in the configmap.yaml and change L87 and L90, too?

Reply:
We discussed this at the sync.

The plan:
We're going to add a json schema file for the chart. We will initially set all the properties that need to be integers as integers in it. This means helm will verify all the input values are integers before it hits the template engine, so the templates don't have to do anything to recheck its an integer or cast. We will do that in a separate PR.

So, in this PR, lets:
remove any ` | int`

And the little bit of the pr here that started the conversation about adding the if's here, lets remove those too.

I think the rest of the pr looks good then.

Reply:
https://github.com/spiffe/helm-charts-hardened/pull/458

Reply:
Cool, thanks for going the extra mile 👍